cadc-vos/src/main/java/ca/nrc/cadc/vos/DataNode.java

```diff
@@ -82,54 +82,75 @@
 public class DataNode extends Node {
     
     private static Logger log = Logger.getLogger(DataNode.class);
-    
-    // Indicates if the node can be accessed.
-    private NodeBusyState busy;
-    
-    // Indicates if the VOSpace understands the data format.
-    private boolean structured;
 
-    /**
-     * Data node constructor.
-     * @param uri
-     */
-    public DataNode(VOSURI uri) {
-        super(uri);
-        this.structured = false;
-        this.busy = NodeBusyState.notBusy;
-    }
+    // The URI to the associated artifact.
+    private final URI storageID;
+
+    // The checksum of the artifact.
+    private final URI contentChecksum;
+
+    // The last modified date of the artifact.
+    private final Date contentLastModified;
+
+    // The content length of the artifact.
+    private final long contentLength;
+
+    // The type of the artifact.
+    public String contentType;
+
+    // The encoding of the artifact.
+    public String contentEncoding;
 
     /**
      * DataNode constructor.
-     * @throws URISyntaxException 
+     *
+     * @param name The name of the node.
+     * @param storageID The URI of associated artifact.
+     * @param contentChecksum The artifact checksum.
+     * @param contentLastModified The artifact lastModified date.
+     * @param contentLength The artifact contentLength.
      */
-    public DataNode(VOSURI uri, List<NodeProperty> properties) throws URISyntaxException {
-        super(uri, properties);
-        this.structured = false;
-        this.busy = NodeBusyState.notBusy;
+    public DataNode(String name, URI storageID, URI contentChecksum, Date contentLastModified, long contentLength) {
+        this(name, storageID, contentChecksum, contentLastModified, contentLength, new TreeSet<>());
     }
 
     /**
-     * @return true if the VOSpace understands the format of the data.
+     * Data node constructor.
+     *
+     * @param name The name of the node.
+     * @param storageID The URI of associated artifact.
+     * @param contentChecksum The artifact checksum.
+     * @param contentLastModified The artifact lastModified date.
+     * @param contentLength The artifact contentLength.
+     * @param properties The node's properties.
      */
-    public boolean isStructured() {
-        return this.structured;
+    public DataNode(String name, URI storageID, URI contentChecksum, Date contentLastModified,
```

Comment:
We do need two ctors, but not one with properties for reasons given in Node. Leave properties out.

Since the Node->Artifact reference is 0..1, there can be a Node without an artifact and hence all the content* fields can be null. The storageID is never null and is immutable. So, you really need a ctor like:
`DataNode(String name, URI storageID)` 
that sets those to null (because final) and the one you have for when there is an artifact, where I'd prefer 
`DataNode(String name, URI contentChecksum, Date contentLastModfied, Long contentLength, URI storageID)` 
since storageID is at the end because it's really internal. And yes, `Long` instead if `long` and a null check.